packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1379,14 +1403,34 @@ export class GarbageCollector implements IGarbageCollector {
                 );
             } else {
                 nodeStateTracker.updateTracking(currentReferenceTimestampMs);
-                if (this.tombstoneMode && nodeStateTracker.state === UnreferencedState.SweepReady) {
-                    const nodeType = this.runtime.getNodeType(nodeId);
-                    if (nodeType === GCNodeType.DataStore || nodeType === GCNodeType.Blob) {
-                        this.tombstones.push(nodeId);
-                    }
+                if (nodeStateTracker.state === UnreferencedState.SweepReady) {
+                    sweepReadyNodes.push(nodeId);
                 }
             }
         }
+
+        return sweepReadyNodes;
+    }
+
+    /**
+     * Removes nodes that are swept from GC state and updates the deleted nodes
+     */
+    private removeDeletedNodesFromGCState(gcNodes: string[]) {
+        assert(this.gcDataFromLastRun !== undefined, "GC data should be defined when we are running sweep");
+        for (const nodeId of gcNodes) {
+            const nodeStateTracker = this.unreferencedNodesState.get(nodeId);
+            // TODO: GC:Validation - assert that the nodeStateTracker is defined
```

Comment:
Can you please create a task for the TODOs so we are tracking it.

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/3248